netbox/vpn/forms/model_forms.py

```diff
@@ -265,9 +265,17 @@ class Meta:
     def __init__(self, *args, initial=None, **kwargs):
         super().__init__(*args, initial=initial, **kwargs)
 
-        if initial and initial.get('type') == TunnelTerminationTypeChoices.TYPE_VIRTUALMACHINE:
+        # If initial or self.data is set and the type is a VIRTUALMACHINE type, swap the field querysets.
+        if (
+            initial and initial.get('type') == TunnelTerminationTypeChoices.TYPE_VIRTUALMACHINE
+        ) or (
+            self.data and self.data.get('type') == TunnelTerminationTypeChoices.TYPE_VIRTUALMACHINE
+        ) or (
+            self.instance.pk and isinstance(self.instance.termination.parent_object, VirtualMachine)
+        ):
```

Comment:
Run-on conditionals like this are not viable from a maintainability standpoint. Please have a look at how we've used the `get_field_value()` utility function in other forms to avoid this.